### PR TITLE
storage: Start lease worker inside a task

### PR DESCRIPTION
**Pull request by bdarnell**
From PR feedback on #5032.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5790)

### CODE REVIEW

**Comment by tamird** [comment]
LGTM

File: storage/replica.go [hunk 1]
```diff
@@ -448,65 +448,70 @@ func (r *Replica) requestLeaderLease(timestamp roachpb.Timestamp) <-chan *roachp
...
-
-			// Send lease request directly to raft in order to skip unnecessary
-			// checks from normal request machinery, (e.g. the command queue).
-			// Note that the command itself isn't traced, but usually the caller
-			// waiting for the result has an active Trace.
-			cmd, err := r.proposeRaftCommand(r.context(context.Background()), ba)
-			if err != nil {
-				return roachpb.NewError(err)
-			}
+	if !r.store.Stopper().RunTask(func() {
```
Comment (spencerkimball) [inline storage/replica.go]: This pattern doesn't seem to make sense to me. Won't this worker never shutdown if it's started in a task?

Could you add a comment about what situation you need to handle with this pattern?

Comment (spencerkimball) [inline storage/replica.go]: Ah, just saw the comment on the stopper and realized that `RunWorker()` starts a goroutine and the caller will exit the task immediately.

How about we add `MaybeRunWorker` to `Stopper` and have it return a boolean indicating whether or not the worker could be started, depending on the status of the stopper.

Comment (bdarnell) [inline storage/replica.go]: I thought about adding a `MaybeRunWorker`, but this pattern is really a hack and not something we want to encourage. It looks like whatever problems you were running into with a task have been fixed (maybe by the context plumbing?), so I think we can just change it to a task: #5873

**Pull request closed without merging**